### PR TITLE
Aaron's pull request

### PR DESCRIPTION
**Pull request by frebbs**

### CODE REVIEW

File: lib/takeaway.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+require_relative './order'
+
+class Takeaway
+  attr_reader :order, :menu_items
+
+  def initialize(menu_items: nil, order: nil)
```
Comment (lindseydew) [inline lib/takeaway.rb]: Lovely use of dependency injection!

File: spec/menuitems_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+require './lib/menuitems'
+
+describe MenuItems do
+
+  subject(:menu) { described_class.new(menuitems) }
+
+  let(:menuitems) do
+    { regina: 12.89, hawaiian: 18.99, bbq_chicken: 12.99, vegan: 14.99, vegetarian: 8.99 }
```
Comment (lindseydew) [inline spec/menuitems_spec.rb]: Nice definition of a variable

File: spec/order_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+describe Order do
+  subject(:order) { described_class.new(items) }
+
+  let(:customer_order) do
+    { bbq_chicken: 3, vegan: 2, hawaiian: 3 }
+  end
+
+  before do
+    allow(items).to receive(:has_item?).with(:bbq_chicken).and_return(true)
```
Comment (lindseydew) [inline spec/order_spec.rb]: Excellent use of mocking here

File: lib/order.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+class Order
+  attr_reader :order, :items
+
+  def initialize(items = nil)
+    @order = {}
+    @items = items || MenuItems.new
+  end
+
+  def add(item, amount)
+    fail "Sorry #{item} is out of stock" unless items.has_item?(item)
```
Comment (lindseydew) [inline lib/order.rb]: Nice!

**Review by lindseydew — COMMENTED** [review]
Excellent job! Your code is well designed, and considers edge cases too with tests.Well done 💯
